src/openrct2-ui/interface/Window.cpp

```diff
@@ -628,11 +628,13 @@ void WindowDrawWidgets(rct_window* w, rct_drawpixelinfo* dpi)
     widgetIndex = 0;
     for (widget = w->widgets; widget->type != WindowWidgetType::Last; widget++)
     {
-        // Check if widget is outside the draw region
-        if (w->windowPos.x + widget->left < dpi->x + dpi->width && w->windowPos.x + widget->right >= dpi->x)
-            if (w->windowPos.y + widget->top < dpi->y + dpi->height && w->windowPos.y + widget->bottom >= dpi->y)
-                WidgetDraw(dpi, w, widgetIndex);
-
+        if (!(widget->flags & WIDGET_FLAGS::IS_HIDDEN))
```

Comment:
Shouldn't we try to use `WidgetIsVisible` everywhere to hide the implementation detail?